### PR TITLE
Add inrobot module

### PR DESCRIPTION
**Pull request by ana-gheorghe**
# Description

We have created an InRobot module to allow customers and/or solution architects to set up InRobot more easily. I've included README files describing the module and what variables are required. 

I have not included a default.config.yaml because I don't really understand how to find the asset external id, the 3D model info (revision/model id, name), and the source ids required for this project. I will need some support from the toolkit team to assemble the default.config.yaml file. We have tested this locally with our own projects and it seems to work as expected.

I will be on vacation for 3 weeks from Aug 4th onwards, but please reach out to the inrobot team (cc @niekbeckers or @atussak) if needed. Otherwise, this can probably wait until I'm back but there's been some request for this from our Japan team for this, so it would be nice to get this available to them.

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/toolkit/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/toolkit/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/cognite_modules/inrobot/cdf_inrobot_common/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+## Robotics
+
+This module creates some robotics specific resources: RobotCapability and DataPostProcessing. 
+
+Currently we support the following RobotCapability resources: acoustic_video, pt_ir_video, ptz_ir, ptz_video, ptz, threesixty, and threesixty_video.
+
+Currently we support the following DataPostProcessing resources:
+process_threesixty, read_dial_gauge, read_digital_gauge, read_level_gauge, read_valve.
+
+If there are any of these robot capabilities you do not want, you can remove the YAML file. Note that the process_threesixty data post processing requires the threesixty capability to be present. Similarly, the various gauge data post processing options require the ptz robot capability.
```
Comment (atussak) [inline cognite_toolkit/cognite_modules/inrobot/cdf_inrobot_common/README.md:39]:
```suggestion
If there are any of these robot capabilities you do not want, you can remove them from the YAML file. Note that the process_threesixty data post processing requires the threesixty capability to be present. Similarly, the various gauge data post processing options require the ptz robot capability.
```

File: cognite_toolkit/cognite_modules/inrobot/cdf_inrobot_common/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## Functions
+
+This module contains four (4) functions that we deploy: Contextualize Robot Data, Gauge Reading, Get IR Data from IR Raw, and ThreeSixty. These are functions that run every minute on files with certain labels that the robot uploads to CDF.
+
+Because the function itself must be stored in a dataset and we use the dataset of the robot, a new function must be defined for each robot in the robots.functions.yaml file. Additionally, most schedules use the robot data set id as a data parameters - this means that a new schedule must be created for every new robot.
+
+## Labels
+
+This module creates two labels that are needed for InRobot to work: robot_gauge and read_ir. These are scoped to the robot's data set. This means that new labels must be created for each additional robot.
```
Comment (atussak) [inline cognite_toolkit/cognite_modules/inrobot/cdf_inrobot_common/README.md:28]: There are some steps that are location or robot specific.
Would it make sense to create a script for "adding a new robot" and one for "adding a new location"?
Just wondering for the future 😁

File: cognite_toolkit/cognite_modules/inrobot/cdf_inrobot_location/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# cdf_inrobot_location
+
+This module contains location specific configurations for InRobot. This is the default location.
+The support multiple locations, copy this module and modify the configurations. Remember to
+rename the module name to e.g. `inrobot_location_<location_name>`.
```
Comment (atussak) [inline cognite_toolkit/cognite_modules/inrobot/cdf_inrobot_location/README.md:5]:
```suggestion
This module contains location specific configurations for InRobot. This is the default location.
To support multiple locations, copy this module and modify the configurations. Remember to
rename the module name to e.g. `inrobot_location_<location_name>`.
```

**Review by atussak — COMMENTED** [review]
This is very smooth 😎

File: cognite_toolkit/cognite_modules/inrobot/cdf_inrobot_common/README.md (hunk 1 above)
Comment (atussak) [reply cognite_toolkit/cognite_modules/inrobot/cdf_inrobot_common/README.md:39]: Oh I see, then it makes sense 😄

**Review by doctrino — APPROVED** [review]
Great job @ana-gheorghe!

Taking the liberty to merge it :)

**Pull request merged**